src/main/java/com/google/idea/perf/methodtracer/ArgSetStats.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.idea.perf.methodtracer
+
+class ArgSetStats(
+    val args: ArgSet,
+    var callCount: Long = 0L,
+    var wallTime: Long = 0L,
+    var maxWallTime: Long = 0L
+)
+
+class ArgStatMap private constructor(
+    val tracepoints: Map<Tracepoint, List<ArgSetStats>>
+) {
+    companion object {
+        fun fromCallTree(root: CallTree): ArgStatMap {
+            val allStats = mutableMapOf<Tracepoint, MutableMap<ArgSet, ArgSetStats>>()
+            val ancestors = mutableSetOf<Tracepoint>()
+
+            fun dfs(node: CallTree) {
+                val nonRecursive = node.tracepoint !in ancestors
+
+                if (node.argSetStats.isNotEmpty()) {
+                    val argSetStats = allStats.getOrPut(node.tracepoint) { mutableMapOf() }
+
+                    for ((args, stats) in node.argSetStats) {
+                        val accumulatedStats = argSetStats.getOrPut(args) {
+                            ArgSetStats(args)
+                        }
+                        accumulatedStats.callCount += stats.callCount
+
+                        if (nonRecursive) {
+                            accumulatedStats.wallTime += stats.wallTime
```

Comment:
Say we're tracing `f` and its first argument. And say `f(1)` calls `f(2)`.

The `f(2)` call will be marked 'nonRecursive', and thus its will `wallTime` will be 0 (right?). Yet I think we do want to account for the wall time spent in `f(2)` (since the tracer will display both `f(1)` and `f(2)` separately).

Reply:
Yup, I will look into it.

Reply:
I think the latest version fixes my example.

However, I have another corner case in mind. Suppose the call tree looks like this:
```
f(1), 10 ms
  f(2), 5 ms
f(2), 5 ms
```
What will be the displayed wall time for `f(2)`?
I think we want it to be `10 ms` but I suspect this code would display `5 ms`.

If I'm right, then I also suspect this would be hard to fix without redesigning how arg values are represented in the call tree. At a minimum we should put in a `TODO` comment.

What do you think?

Reply:
Yeah, I have to look more into this too.